src/mds/Locker.cc

```diff
@@ -5430,10 +5459,39 @@ void Locker::handle_file_lock(ScatterLock *lock, const cref_t<MLock> &m)
     } else {
       dout(7) << "handle_file_lock trying nudge on " << *lock
 	      << " on " << *lock->get_parent() << dendl;
-      scatter_nudge(lock, 0, true);
+      if (lock->get_type() == CEPH_LOCK_INEST) {
+        auto p = m->get_data().cbegin();
+        utime_t rstat_dirty_from;
+        decode(rstat_dirty_from, p);
+        CInode* ino = static_cast<CInode*>(lock->get_parent());
+        if (ino->get_projected_inode()->rstat_dirty_from.is_zero()
+            || (!rstat_dirty_from.is_zero() &&
+              rstat_dirty_from < ino->get_projected_inode()->rstat_dirty_from))
+          ino->get_projected_inode()->rstat_dirty_from = rstat_dirty_from;
```

Comment:
modify rstat_dirty_from without projecting inode is wrong. I'm wondering why this code is required? why encoding rstat_dirty_from in LOCK_ACK_FOO message is not enought